Dh 10

Dyrk84 opened the pull request.

Dyrk84 committed:
DH-9
Renamed the same values names in Champion and Monster classes.
Added 4 new unit test (champion and monster win tests, with fail result too.
New fight end code, gameOver method removed.

Dyrk84 committed:
DH-10
Monster hp initializing with Dice.rollDice method.
Now the monsters number is random(with the Dice.rollDice method), if a monster die, a new one comes until they all die.
If all monsters are dead, prints a victory message.
Reworked championAttack and monsterAttack methods.

Dyrk84 committed:
DH-10
próba unstash

Dyrk84 committed:
DH-10
Reworked code beginning, afterMain method is now in Fight constructor.
new lombok setters and getters in Champion and Monsters class.

gergelyszanto commented on src/main/java/hu/dungeonhunter/characters/Champion.java:
```diff
@@ -1,52 +1,50 @@
 
-    public int getChampionMaxDamage() {
-        return championMaxDamage;
+    void startValues() {
```
Keep methods private if not accessed from other classes.

gergelyszanto commented:
Two of your tests are failing:
![image](https://user-images.githubusercontent.com/24583442/108272160-2a9bdc00-7172-11eb-9f22-246ab98bec2f.png)

gergelyszanto commented on src/main/java/hu/dungeonhunter/tools/Fight.java:
```diff
@@ -1,47 +1,100 @@
-            gameOver();
+    public Fight(){
+        monsterCounter = Dice.rollDice(6,2);
+        System.out.println(monsterCounter + " monsters is in the Dungeon!");
```
```suggestion
        System.out.println(monsterCounter + " monsters are in the Dungeon!");
```

Dyrk84 replied:
Thank you, i corrected!

gergelyszanto commented on src/main/java/hu/dungeonhunter/tools/Fight.java:
```diff
@@ -1,47 +1,100 @@
+
+    public Monsters monsterCaller() {
+        if (monsterCounter != 0) {
+            System.out.println("A monster step out from darkness!");
```
```suggestion
            System.out.println("A monster steps out from darkness!");
```

Dyrk84 replied:
Thank you, i corrected this too!

gergelyszanto commented on src/main/java/hu/dungeonhunter/tools/Fight.java:
```diff
@@ -1,47 +1,100 @@
+        if (monsterCounter != 0) {
+            System.out.println("A monster step out from darkness!");
+            monster = new Monsters();
+        }return monster;
```
```suggestion
        }
        return monster;
```

Dyrk84 replied:
Does this need to make the code look better?
I corrected.

gergely-szanto-epam replied:
Basically yes. Easier to see the if block and the return value if separated.

gergelyszanto commented on src/main/java/hu/dungeonhunter/tools/Fight.java:
```diff
@@ -1,47 +1,100 @@
+        start();
+    }
+
+    public void start(){
```
Let's rename this to spawnMonster or something like that. start is not a verbose name (looking at the method name you don't know what it does start)

gergelyszanto commented on src/main/java/hu/dungeonhunter/tools/Fight.java:
```diff
@@ -1,47 +1,100 @@
+    }
+
+    int chosenNumber() {
+        do {
+            Scanner scanner = new Scanner(System.in);
+            try {
+                return scanner.nextInt();
+            } catch (InputMismatchException hibafogo) {
+                yourChooseIsNotAppropriate();
+                printMenu();
+            }
+        } while (true);
```
```suggestion
        do {
            printMenu();
            Scanner scanner = new Scanner(System.in);
            try {
                return scanner.nextInt();
            } catch (InputMismatchException hibafogo) {
                yourChooseIsNotAppropriate();
            }
        } while (true);
```
And remove printMenu(); from line 93

Dyrk84 replied:
Why? For dont take 2 line the printMenu method ?
I corrected.

gergely-szanto-epam replied:
Yes, it is enough to call printMenu at one place. When the user is asked to enter an input, the menu will be automatically printed for the user.

gergelyszanto commented:
Add test to validate monster counter (monster defeat)

gergelyszanto commented on src/main/java/hu/dungeonhunter/characters/Champion.java:
```diff
@@ -1,52 +1,50 @@
+        this.maxDamage = 6;
+        this.numOfDices = 1;
+        startValues();
+        this.defeat = false;
```
No need to give false initializer value to a boolean. When you say `private boolean defeat;` at line 23 it already has false value.

gergelyszanto commented on src/main/java/hu/dungeonhunter/characters/Monsters.java:
```diff
@@ -1,51 +1,47 @@
+        this.maxDamage = 6;
+        this.numOfDices = 1;
+        startValues();
+        this.defeat = false;
```
Same here as at champion, no need for false initializer value.

Dyrk84 committed:
DH-10
Prints the menu in the chosenNumber method.
The constructor method is private now in champion and monster class.
New monsterNumber and killedMonsterNumber tests.

gergelyszanto committed:
DH-10 kontarkodas

gergelyszanto committed:
DH-10 kontarkodas

Dyrk84 committed:
DH-10
added ChampionDefeatTest

Dyrk84 committed:
Merge remote-tracking branch 'origin/DH-10' into DH-10
